packages/next/build/webpack/plugins/middleware-plugin.ts

```diff
@@ -686,6 +686,24 @@ function getExtractMetadata(params: {
       for (const module of modules) {
         const buildInfo = getModuleBuildInfo(module)
 
+        /**
+         * Check if it uses the image generation feature.
+         */
+        if (!dev) {
+          const resource = module.resource
+          if (
+            resource &&
+            /[\\/]node_modules[\\/]@vercel[\\/]og[\\/]dist[\\/]index.js$/.test(
+              resource
+            )
+          ) {
+            telemetry.record({
+              eventName: 'NEXT_EDGE_IMAGE_GENERATION_USED',
```

Comment:
Should this be under `EVENT_BUILD_FEATURE_USAGE` x-ref: https://github.com/vercel/next.js/blob/5e55c3c340c7f6bb25aebb161e554db3cbfcef90/packages/next/telemetry/events/build.ts#L133

Note this also needs to be added to the backend schema for it to be accepted.

Reply:
Thanks for the pointers!

Reply:
@ijjk Added https://github.com/vercel/next-telemetry/pull/95 and updated this PR.